flyteplugins/go/tasks/pluginmachinery/flytek8s/pod_helper_test.go

```diff
@@ -2090,19 +2090,16 @@ func TestMergeWithBasePodTemplate(t *testing.T) {
 func TestMergePodSpecs(t *testing.T) {
 	var priority int32 = 1
 
-	podSpec1, _ := MergePodSpecs(nil, nil, "foo", "foo-init")
+	podSpec1, _ := MergeBasePodSpecOntoTemplate(nil, nil, "foo", "foo-init")
 	assert.Nil(t, podSpec1)
 
-	podSpec2, _ := MergePodSpecs(&v1.PodSpec{}, nil, "foo", "foo-init")
+	podSpec2, _ := MergeBasePodSpecOntoTemplate(nil, &v1.PodSpec{}, "foo", "foo-init")
 	assert.Nil(t, podSpec2)
 
-	podSpec3, _ := MergePodSpecs(nil, &v1.PodSpec{}, "foo", "foo-init")
-	assert.Nil(t, podSpec3)
-
 	podSpec := v1.PodSpec{
 		Containers: []v1.Container{
 			v1.Container{
-				Name: "primary",
```

Comment:
`primary` and `primary-init` should be reserved for templates. Renamed these since I fixed the ordering of args passed to the merge function.